lib/model.js

```diff
@@ -2884,7 +2884,7 @@ class Model {
         build.set(this._timestampAttributes.updatedAt, values[this._timestampAttributes.updatedAt], { raw: true });
 
         if (options.sideEffects) {
-          values = _.assign(values, _.pick(build.get(), build.changed()));
+          values = _.assign(values, _.pick(build.dataValues, build.changed()));
```

Comment:
`.get()` should be fixed to return correct representation of instance data. What does it return for incorrect case?

Reply:
I think that what `.get()` return is not important.
`build.changed()` returns changed `.dataValues` so we should use `.dataValues` in this case.
https://github.com/sequelize/sequelize/blob/d18c2d3d1b9765e10fc804772e28d28773847da1/lib/model.js#L3533

Reply:
`build.changed()` return keys using `dataValues` correct, but we still want `values` to take care of getters. 

What does `build.get()` return for incorrect case?

Reply:
> but we still want values to take care of getters

Could you please tell me the reason? 

> What does build.get() return for incorrect case?

Please let me know the intention of your question. I'm sorry that I cannot understand what you mean.

Reply:
@sushantdhiman I would appreciate your reply.

Reply:
Hi @u9r52sld sorry for this late reply (as usual :smile:), so I checked original issue and code today, here is what I meant earlier

**Reference attribute definition**

```js
geo: {
  type: DataTypes.GEOMETRY('POINT'),
  allowNull: false,
  get() {
    return this.getDataValue('geo').coordinates;
  },
  set(coordArr) {
    this.setDataValue('geo', {
      type: 'Point',
      coordinates: coordArr
    });
  }
}
```

1. Error appears because `instance.get()` fails
2. `get` fails because user didn't take into account that `getDataValue('field')` can return `undefined` when using `getter` for an attribute
3. `getDataValue('field')` returns `undefined` because `Model.build` didn't set any value, it can;t as user didn't passed any value for `geo` in `update` payload

Originally I meant that `get` should be fixed, but it appears `get` implementation for this attribute is failing, I don;t think we can handle that other than properly pointing out which attribute getter call has failed.

What do you think?